maliput-sys/src/math/math.h

```diff
@@ -0,0 +1,79 @@
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#pragma once
+
+#include <memory>
+
+#include "maliput/math/vector.h"
+
+#include <rust/cxx.h>
+
+namespace maliput {
+namespace math {
+
+/// Creates a new maliput::math::Vector3.
+/// Forwads to maliput::math::Vector3(double x, double y, double z) constructor.
+std::unique_ptr<Vector3> Vector3_new(rust::f64 x, rust::f64 y, rust::f64 z) {
+  return std::make_unique<Vector3>(x, y, z);
+}
+
+/// Returns the norm of the vector.
+/// Forwards to maliput::math::Vector3::norm() method.
+double Vector3_norm(const Vector3& self) {
+  return self.norm();
+}
+
+/// Returns the dot product of two vectors.
+/// Forwards to maliput::math::Vector3::dot(const Vector3& other) method.
+double Vector3_dot(const Vector3& self, const Vector3& other) {
+  return self.dot(other);
+}
+
+/// Returns the cross product of two vectors.
+/// Forwards to maliput::math::Vector3::cross(const Vector3& other) method.
+std::unique_ptr<Vector3> Vector3_cross(const Vector3& self, const Vector3& other) {
+  return std::make_unique<Vector3>(self.cross(other));
+}
+
+/// Returns true if the two vectors are equal.
+/// Forwards to maliput::math::Vector3::operator==(const Vector3& other) method.
+bool Vector3_equals(const Vector3& self, const Vector3& other) {
+  return self == other;
+}
+
+/// Returns the string representation of the vector.
+/// Forwards to maliput::math::Vector3::to_str() method.
+rust::string Vector3_to_str(const Vector3& self) {
```

Comment:
I checked in [here](https://github.com/dtolnay/cxx/blob/master/tests/ffi/tests.cc#L334-L338) and it is curious to see `rust::String` instead of `rust::string`... 
Also, the docs refer to `rust::String` : https://docs.rs/cxx/latest/cxx/#builtin-types

Can you help me understanding which is which?

Reply:
There is an alias for it in cxx.h header file: `using string = String;`
https://github.com/dtolnay/cxx/blob/master/include/cxx.h#L467-L469

But yeah it is just a mere decision on the guidelines we use/want. Probably `rust::String` is more aligned with our classes style.

Reply:
Done